File: app-geekhome/src/main/java/eu/geekhome/rest/plugins/PluginDto.java
package eu.geekhome.rest.plugins;

import com.geekhome.common.localization.Resource;
import com.google.gson.annotations.SerializedName;

public class PluginDto {

    @SerializedName("provider")
    private final String _provider;

    @SerializedName("version")
    private final String _version;

    @SerializedName("enabled")
    private final boolean _enabled;

    @SerializedName("name")
    private final Resource _name;

    @SerializedName("description")
    private final Resource _description;

    @SerializedName("id")
    private final String _id;

    @SerializedName("isHardwareFactory")
    private final boolean _isHardwareFactory;

    public PluginDto(String id, Resource name, Resource description, String provider, String version, boolean isHardwareFactory, boolean enabled) {
        _name = name;
        _description = description;
        _id = id;
        _provider = provider;
        _version = version;
        _isHardwareFactory = isHardwareFactory;
        _enabled = enabled;
    }

    public String getProvider() {
        return _provider;
    }

    public String getVersion() {
        return _version;
    }

    public boolean isEnabled() {
        return _enabled;
    }

    public Resource getName() {
        return _name;
    }

    public Resource getDescription() {
        return _description;
    }

    public String getId() {
        return _id;
    }
}

File: modules/data/src/main/java/eu/automateeverything/data/plugins/PluginDto.kt
/*
 * Copyright (c) 2019-2022 Tomasz Babiuk
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  You may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package eu.automateeverything.data.plugins

import eu.automateeverything.data.localization.Resource
import eu.automateeverything.data.settings.SettingGroupDto
import kotlinx.serialization.Serializable

@Serializable
data class PluginDto(
    val id: String,
    val name: Resource,
    val description: Resource,
    val copyright: Resource?,
    val category: PluginCategory,
    val provider: String,
    val version: String,
    val isHardwareFactory: Boolean,
    val enabled: Boolean,
    val settingGroups: List<SettingGroupDto>
)